conda_build/utils.py

```diff
@@ -68,6 +68,16 @@ def tar_xf(tarball, dir_path, mode='r:*'):
         subprocess.check_call([uncompress, '-f', tarball])
         tarball = tarball[:-2]
     if not PY3 and tarball.endswith('.tar.xz'):
+        # try extracting directly with tar,
+        # which may be able to cope with xz compression
+        tar = external.find_executable('tar')
+        if tar:
+            try:
+                subprocess.check_call([tar, '-x', '-C', dir_path, '-f', tarball])
+                return
+            except:
```

Comment:
Please only catch the specific exception (`subprocess.CalledProcessError` I believe).